packages/asset-server-plugin/src/s3-asset-storage-strategy.ts

```diff
@@ -157,113 +157,114 @@ export function configureS3AssetStorage(s3Config: S3Config) {
  * @docsWeight 0
  */
 export class S3AssetStorageStrategy implements AssetStorageStrategy {
-    private AWS: typeof import('aws-sdk');
-    private s3: import('aws-sdk').S3;
+    private AWS: typeof import('@aws-sdk/client-s3');
+    private s3: import('@aws-sdk/client-s3').S3;
     constructor(
         private s3Config: S3Config,
         public readonly toAbsoluteUrl: (request: Request, identifier: string) => string,
     ) {}
 
     async init() {
         try {
-            this.AWS = await import('aws-sdk');
-        } catch (e: any) {
+            this.AWS = await import('@aws-sdk/client-s3');
+        } catch (err: any) {
             Logger.error(
                 'Could not find the "aws-sdk" package. Make sure it is installed',
                 loggerCtx,
-                e.stack,
+                err.stack,
             );
         }
 
         const config = {
-            credentials: this.getS3Credentials(),
+            credentials: await this.getS3Credentials(),
             ...this.s3Config.nativeS3Configuration,
-        };
+        } satisfies S3ClientConfig
+
         this.s3 = new this.AWS.S3(config);
         await this.ensureBucket(this.s3Config.bucket);
     }
 
     destroy?: (() => void | Promise<void>) | undefined;
 
     async writeFileFromBuffer(fileName: string, data: Buffer): Promise<string> {
-        const result = await this.s3
-            .upload(
-                {
-                    Bucket: this.s3Config.bucket,
-                    Key: fileName,
-                    Body: data,
-                },
-                this.s3Config.nativeS3UploadConfiguration,
-            )
-            .promise();
-        return result.Key;
+        const upload = new (await import('@aws-sdk/lib-storage')).Upload({
```

Comment:
Do you think it makes sense to also import the `Upload` class in the `init()` method so it does not need to be done every time these methods are called?

Reply:
Did not think about it 😅 will change

Reply:
Will answer in a comment to the main PR